### DISCUSSION TITLE
Vault Automatic Sync

### DISCUSSION THREAD

**Discussion by attzonko** [dani-garcia/vaultwarden#1455] [Q&A]
I love self-hosting Bitwarden_rs, I also love the Duo 2FA option. 

There is one item that I can't figure out: Is there a way to enable automatic synchronization across all the different entities?

For example, if I add a new password using the Chrome extension, should I expect to see that new password in my vault after some time, or do I manually have to go through the extension menu items and force the sync?

Or similarly, if I add a new password through the web front end, when can I expect to see it available in the desktop app running on my other computer?

Does the manual forced sync do a merge operation?

**Comment by mqus** [comment]
Afaik, if you enter a password in any client, it will be immediately synced to the server. If it cannot connect to the server, you can't add new passwords, only access existing ones. This should prevent merging issues. How the sync works in the direction server->client I don't know. But I suspect that it is pretty much immediate if the websocket connection is used.
